metrics/src/metrics.rs

```diff
@@ -20,8 +20,8 @@ use {
 
 type CounterMap = HashMap<(&'static str, u64), CounterPoint>;
 
-impl From<CounterPoint> for DataPoint {
-    fn from(counter_point: CounterPoint) -> Self {
+impl From<&CounterPoint> for DataPoint {
+    fn from(counter_point: &CounterPoint) -> Self {
```

Comment:
The original way should be a `move` instead of a copy as it does not implement the copy trait.  Do these lines of change also contribute to the performance gain or just from lines 165-166?

Reply:
seems to me like this is probably the bulk of the improvement. should be like 5x less memcpy'd.  the drain+push is nearly equivalent to the extend+map without it. might be able to improve drain+psuh that with a call to `reserve()` before the loop

Reply:
Thanks for the explanation.  I originally thought `move` can be done by simply moving the ownership of the struct, but  Rust does take care of the ownership of each member field independently.